internal/statemachine/classic_states.go

```diff
@@ -318,65 +317,92 @@ func (stateMachine *StateMachine) installPackages() error {
 			classicStateMachine.ImageDef.Kernel)
 	}
 
-	// Slice used to store all the commands that need to be run
-	// to install the packages
+	// Slice used to store all the commands that need to be run to install the packages
+	// installPackagesCmds should be filled as a FIFO list
 	var installPackagesCmds []*exec.Cmd
 
-	// mount some necessary partitions from the host in the chroot
-	type mountPoint struct {
-		dest     string
-		fromHost bool
-	}
+	// Slice used to store all the commands that need to be run
+	// to properly cleanup everything after the packages installation
+	// teardownCmds should be filled as a LIFO list (so new entries should added at the start of the slice)
+	var teardownCmds []*exec.Cmd
+
+	// This will take care of tearing down everything as much as possible
+	// It will be executed even if the function paniced and will continue if one of the command failed
+	// to left the system as clean as possible if something has gone wrong
+	defer func() {
```

Comment:
I don't know the coding conventions in Go as much, but I was wondering if the teardown code of iterating through the `teardownCmd` 's and printing errors could be split out to a separate function? Since the code seems to be basically the same whenever it's used, so it's a bit of a boilerplate code.

Reply:
It is a bit impractical here because the body of the defered function is referencing variables defined in the parent (here installPackages) function. This is done on purpose because if we give `teardownCmds` and `err` as arguments to the defered functions, these arguments are evaluated immediately, but we want them to be evaluated when the defered function run. 
But I think I found a solution to workaround that, I will try it.